### ISSUE TITLE
💌📯 Updates from semantic-release

### ISSUE THREAD

**Issue by gr2m** [semantic-release/semantic-release#411]
👋 Hello friends, my name is Gregor and my friend Stephan @boennemann invited me as maintainer to semantic-release to help move it forward. Some of you might now me from [Hoodie](http://hood.ie/), that’s the project that semantic-release was originally developed for before it became it’s own thing. We use it in all our repositories there and I personally use `semantic-release` in _all_ of my personal projects. 

Honestly, without semantic-release, I would not be able to maintain nearly as many Open Source projects as I do today.

Please subscribe to this issue for updates. I will lock the issue to keep it for updates only, but trust me I know most of your pains myself and I hear you.

I will be very busy in August but will have more time starting September as I [transition into freelancing](https://twitter.com/gr2m/status/895690164168572928) :)

**If you like to help** <3 triaging the existing issues and pull-requests and pointing folks to this issue for updates would be great ✨ If you find issues that you think I should look into next, please mention me in them and I’ll try to have a look. 

Thank for your patience with me! Expect some great updates coming your way :)

**Comment by gr2m** [comment]
The fix of the `"<a name"x.x.x"> in release notes"` bug is currently blocked by a problem with npm publish: https://gist.github.com/gr2m/80e7e580556c1f7eb32d7b8af55ff281

**Comment by boennemann** [comment]
Sorry for neglecting semantic-release in the past and thanks @gr2m for stepping in.
Happy to see progress in here 💖

**Comment by gr2m** [comment]
I’ve released `semantic-release@7.0.0` with the next release. You can install it with `npm install --save-dev semantic-release@next` or directly set the version in your projects’ `package.json` files. 

Note that support for no longer maintained Node.js versions has been dropped as part of this release. If you use Node 4 or newer all will be fine :) I’m very sorry if you still have run on Node 0.10 or Node 0.12, but we can’t support these any longer while at the same time making the project easier to maintain and contribute to.

This should resolve the `"<a name"x.x.x"> in release notes"` issue. Please give it a try and let me know if it works as expected, once a few folks tested it we can set the `semantic-release@7.0.0` to `@latest`

**Comment by gr2m** [comment]
I have set `semantic-release@7.0.1` to `@latest`. Happy updating :)

**Comment by gr2m** [comment]
We had a little boo boo in `v7.0.1`. If a new version was released using that version (past two days), please see the notes on how to recover from the `ENOTINHISTORY ` error at https://github.com/semantic-release/semantic-release/releases/tag/v7.0.2

Sorry for that 🙇

**Comment by gr2m** [comment]
We just released `semantic-release@8` with the `@next` dist-tag. It drops support for Node < 8:
https://github.com/semantic-release/semantic-release/releases/tag/v8.0.0

Install the latest version with `npm install --save-dev semantic-release@next` or update the version in your package.json file directly. We would much appreciate your help testing it before we set it to `@latest` 🙏 

With this version we also introduced a new way to detect the Travis build leader, meaning the job on which semantic-release will be run. Before, it was run on the first job, now we look for the first job that is running on node@8 instead. If you have a build matrix that includes version 8, you are all set, if not, add node 8 to your test matrix (you should do that anyway, it is the current stable version of node).

In case you cannot add node 8 to your test matrix, please let us know why in a new issue (mention me please). To workaround, you can install [npx](https://www.npmjs.com/package/npx) and then run `npx -p node@8 npm run semantic-release` in `after_success`.

**Comment by gr2m** [comment]
semantic-release got a nice shoutout on [🎙️ JS Party #18](https://changelog.com/jsparty/18). Thanks @mikeal ✨

**Comment by gr2m** [comment]
Pierre (@vanduynslagerp) offers to donate two of his plugins he wrote to semantic-release core:
https://github.com/semantic-release/semantic-release/issues/427

Let us know what you think and if you could help us give the plugins a try to make sure they don’t break existing behaviour by enabling them in your projects, that’d be great :)

**Comment by gr2m** [comment]
We are moving forward with @vanduynslagerp plugins. Pierre will also join the maintainers team 🙌

**Comment by gr2m** [comment]
I’ve set semantic-release@8.0.0 to the `@latest` dist-tag on version as no problems have been reported in the past 4 weeks :) Happy updating ✨

**Comment by vanduynslagerp** [comment]
We just released `semantic-release@8.0.1` `@next` dist-tag. 🎉

It includes 2 refactored default plugins: [commit-analyzer](https://github.com/semantic-release/commit-analyzer) and [release-notes-generator](https://github.com/semantic-release/release-notes-generator).

Among the new features:
* Option to select commit message style via [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog) `preset` or `config`. For example you can choose to use the [eslint](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint) commit format rather than the default `angular`.
* Option to customize the commit analysis rules with criteria via [release-rules](https://github.com/semantic-release/commit-analyzer#release-rules) option. For example you can configure the commits with a `docs` type and a `readme` scope to trigger a `patch` release. Criteria can also use regex and can be defined on any field parsed by the selected [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog) `preset`.
* Allow to overwrite specific [conventional-changelog](https://github.com/conventional-changelog/conventional-changelog) `preset` options via `parserOpts` and `writerOpts `. It's convenient if you want to use an existing `preset` but with a small difference, for example a more loose parser regex or using `type[scope]` instead of `type(scope)`.

These new featuree should address several issues in the tracker requesting more flexibility regarding the commit format.

You can can try those new feature by installing `semantic-release` with `npm install semantic-release@next`.
Let us know what you think ! Feedback would be very much appreciated !

**Comment by gr2m** [comment]
We just released `semantic-release@8.0.3` on `@latest`. [release notes](https://github.com/semantic-release/semantic-release/releases)

**Comment by pvdlg** [comment]
We've just release `semantic-release@8.1.0` on the `@next tag`.
You can test it by updating your `package.json` explicitly referencing the new version:
```json
"devDependencies": {
  "semantic-release": "^8.1.0"
}
```
Or by running `npm install --save-dev semantic-release@next`.

This new release ship with #453 that change the way we determine the git head of the last release which we use to determine the commits to analyze since the last version.

We used to rely on the `gitHead` filed in the package metadata on the npm registry, but under certain circumstances this field might not be published on npm, resulting in the infamous `ENOTINHISTORY` error on `semantic-release`.

In `semantic-release@8.1.0`, when the `gitHead` is missing from the npm registry metadata we check if there is a git tag associated with the version of the last release and if there is one we use the associated commit.
In addition we make sure to unshallow the git repository cloned by the CI, when necessary, to be sure to have access to all git tags and commits.

Please give it a try and let us know if it works as expected. You can comment on #453 if you encounter any issue or have a suggestion for improvement.

**Comment by gr2m** [comment]
We just released `semantic-release@8.2.0` as `@latest`. Last `@latest` release was 8.0.3. See [release notes](https://github.com/semantic-release/semantic-release/releases)

**Comment by pvdlg** [comment]
We are discussing about improving modularization and allowing to cover more workflow in the issue #484.
It will be a pretty important change in `semantic-release` and should allow to cover many feature requested.
We'd love to have some feedback from the community ! Feel free to check #484 and help us to find the best solution.

**Comment by pvdlg** [comment]
We just released `semantic-release@11.0.0` on the `next` dist-tag. It comes with different optional way to set options that doesn't rely on `package.json`. This open the door to using `semantic-release` for any language / type of project via plugins.

Be aware that plugins now have to return an async function instead of a function accepting a callback.

The plugin architecture shouldn't change much now. So if you'd like to develop a plugin for your favorite package manager, git repo etc...it's safe to start now!

The next steps I'm going to work on:
- A plugin to release via commit (allow to release commit dist files, Changelog etc...)
- A plugin to run shell scripts
- A way to handle multiple branches / dist-tag / pre-release, in a fully automated way!

**Comment by pvdlg** [comment]
I did some cleaning/sorting/labeling on the issues/PR across the `semantic-release` repos.

A lot of issues a labeled with [help wanted](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+user%3Asemantic-release+archived%3Afalse+label%3A%22help+wanted%22). Feel free to check them out and add a comment if you'd like to help and tackle some. Thanks !

**Comment by pvdlg** [comment]
We just `released semantic-release@11.0.2` as @latest.

#### New features
- `Semantic-Release` must now be executed with `semantic-release` instead of `semantic-release pre && npm publish && semantic-release post`
- The CLI options `--debug` doesn't enable the dry-run mode anymore but activate the debug logs. The dry run mode is now set with the CLI command `--dry-run` or `-d`.
- [npm](https://github.com/semantic-release/github) and [github](https://github.com/semantic-release/github) releases are now handled within their one plugins which can be disabled. The plugins also offer a lot of new options
- semantic-release can now be used for non `npm` packages and can be configured via dotfiles as an alternative to `package.json` 

See the [release notes](https://github.com/semantic-release/semantic-release/releases) for more details.

#### Migration guide

If you are using a third-party plugin, please verify its documentation and repository to make sure it has been updated to work with semantic-release version `11.0.0`.

Update the semantic-release in your package.json from:
```json
"scripts": {
   "semantic-release": "semantic-release pre && npm publish && semantic-release post"
}
```
To:
```json
"scripts": {
   "semantic-release": "semantic-release pre && npm publish && semantic-release post"
}
```

**Comment by pvdlg** [comment]
The next step will be support more complex release workflows, including multiple branches, distribution channels, Long term support branches and pre-releases.

Feedback would be much appreciated in #563 !

**Comment by pvdlg** [comment]
semantic-release 12 has been released on the `latest` dist-tag!
It comes with changes regarding the CI configuration.

Please see the [release notes](https://github.com/semantic-release/semantic-release/releases/tag/v12.0.0) for more details and a migration guide.

**Comment by pvdlg** [comment]
I'm closing as it's not used anymore for a while.
The project progress can be tracked here: https://waffle.io/semantic-release/semantic-release
